docs/spec/consensus/fork-detection.md

```diff
@@ -0,0 +1,93 @@
+# Fork Detection and processing of misbehavior with lite clients
+
+As explained in [fork-accountability](fork-accountability.md) document we say that a fork is a case in which an honest process (validator, full node or lite client)
+observed two commits for different blocks at the same height of the blockchain. With Tendermint consensus protocol, a fork can only
+happen if [Tendermint-Failure-Model](light-client.md#tendermint-failure-model) does not hold, i.e., we have more than 1/3 of voting power under control
+of faulty validators in the relevant validator set.
+In this document we focus on detections of forks observed only by lite clients, i.e., we assume that there are no forks 
+on the main chain, and that a fork is used as an attack to some lite client.
+
+There are several sub-problems we will cover in this document:
+
+- a problem of detecting a fork by an honest lite client,
+  
+- identifying faulty validators (and not mistakenly accusing correct validators) and
+  
+- processing misbehavior on the main chain (if possible).  
+
+## Detecting forks
+
+A fork is defined as a deviation (of list of blocks) from the main chain. In this document we will assume that there exists 
+a main chain *C1* (sequence of blocks) and an honest full node *F* that is synced to the chain *C1*. Then a fork corresponds 
+to an existence of a commit for block *B* for height *h* that is different from a block at height *h*
+on the chain *C1*. Furthermore, a commit for the block *B* is a *valid* fork if it contains at least a single valid signature 
+by a validator from the validator set at height *h* at chain *C1*. This definition of valid fork is needed to differentiate
+forks that can provide material for punishment mechanisms (slashing) from bogus data that looks like a valid commit
+but is signed by processes that are not tracked by the system (and therefore can't be punished).
+
+As explained in [lite-client](light-client.md) document a lite client establish a connection with a full node and it verifies application states 
+of interest by downloading and verifying corresponding block headers. To be able to detect a fork, we assume an additional 
+module (fork detection module) that runs in parallel with the core lite client module, and whose role is fork detection. 
+The fork detection module maintains a list of peers (full nodes) and it periodically establishes a connection to some peer 
```

Comment:
"periodically" do we need to make this configurable? or place restrictions like verify each header at least by 1 witness?

Reply:
I am not sure yet what is the best way to formalise that part. Probably the best would be to make it configurable so user can choose what makes most sense for them.